google-cloud-spanner-executor/src/main/java/com/google/cloud/executor/spanner/CloudExecutorImpl.java

```diff
@@ -48,6 +56,33 @@ public StreamObserver<SpannerAsyncActionRequest> executeActionAsync(
       @Override
       public void onNext(SpannerAsyncActionRequest request) {
         LOGGER.log(Level.INFO, String.format("Receiving request: \n%s", request));
+
+        // Use Multiplexed sessions for all supported operations if the
+        // multiplexedSessionOperationsRatio from command line is > 0.0
+        if (multiplexedSessionOperationsRatio > 0.0) {
+          SessionPoolOptions.Builder sessionPoolOptionsBuilder;
+          if (request.getAction().getSpannerOptions().hasSessionPoolOptions()) {
+            sessionPoolOptionsBuilder =
+                request
+                    .getAction()
+                    .getSpannerOptions()
+                    .getSessionPoolOptions()
+                    .toBuilder()
+                    .setUseMultiplexed(true);
+          } else {
+            sessionPoolOptionsBuilder = SessionPoolOptions.newBuilder().setUseMultiplexed(true);
+          }
+
+          SpannerOptions.Builder optionsBuilder =
+              request
+                  .getAction()
+                  .getSpannerOptions()
+                  .toBuilder()
+                  .setSessionPoolOptions(sessionPoolOptionsBuilder);
+          SpannerAction.Builder actionBuilder =
+              request.getAction().toBuilder().setSpannerOptions(optionsBuilder);
+          request = request.toBuilder().setAction(actionBuilder).build();
```

Comment:
Since we don't have unit tests, can you please link a before/after log object of this request object before and after your change? This is to ensure we are not mis-constructing the entire input object and avoid bugs like above.

Reply:
I have added a short log [here](https://paste.googleplex.com/6423074894774272).

**Request object Before:**
<img width="597" alt="Screenshot 2024-04-23 at 3 19 16 PM" src="https://github.com/googleapis/java-spanner/assets/57220027/59fb5244-b948-405d-9b2e-eef75fd5ad7f">

Request Object after adding multiplexed session
<img width="597" alt="Screenshot 2024-04-23 at 3 24 38 PM" src="https://github.com/googleapis/java-spanner/assets/57220027/ff767fda-6e00-4624-aadb-c888782f1ec4">